052-record-reconciled-versions.md

```diff
@@ -0,0 +1,198 @@
+  Once all other components are installed, the `Kafka` CR updates to:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.37.0
+  ```
+- User can see `metadata.annotations[strimzi.io/reconciled] = 0.37.0` so they know it's reconciled to correct level
+
+### Upgrade (with new mechansim)
+This defines an example where it is imagined that 0.37.0 had this mechanism already implemented, and is now upgrading to a new version 0.38.0
+
+- On an upgrade from 0.37.0, a user has a Kafka CR:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.37.0
+  ```
+- The Kafka Reconciler watcher picks up the CR and does an initial status update
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.38.0
+  ```
+  and updates `StrimziPodSet` CR (not changing the annotations from previous, since Kafka reconciler does not own them) with:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.37.0
+  ```
+- StrimziPodSet reconciler reconciles the SPS CR with annotations:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.38.0
+  ```
+  and starts deploying Kafka brokers one at a time, and on completion updates the SPS CR to:
+  ```
+  strimzi.io/reconciled: 0.38.0
+  strimzi.io/reconciling: 0.38.0
+  ```
+- Kafka Reconciler picks up that SPS is now `Ready` and finished, installing other components
+  Once all other components are installed, the `Kafka` CR updates to:
+  ```
+  strimzi.io/reconciled: 0.38.0
+  strimzi.io/reconciling: 0.38.0
+  ```
+- User can see `metadata.annotations[strimzi.io/reconciled] = 0.38.0` so they know the CR has been reconciled to correct level
+
+
+## Kafka Version Annotation
+As a follow-on from this proposal, the mechanism could be extended to cover Kafka versions as well, such as:
+```
+strimzi.io/kafkaVersion: 3.5.0
+strimzi.io/kafkaVersionReconciling: 3.6.0
+```
+Where the `strimzi.io/kafkaVersion` annotation is updated only once the SPS has rolled all brokers to the new version.
+And `strimzi.io/kafkaVersionReconciling` signals that a Kafka version update is in progress if `kafkaVersionReconciling` and `kafkaVersion` don't match.
+
+## Affected/not affected projects
+To the best of my knowledge only the `strimzi-kafka-operator` would be effected by this mechanism, including the code changes to the reconcilers, the doc changes, and the updates to the system tests.
+
+## Compatibility
+
+This mechanism is backwards compatible, in the sense that it works even if the annotations were not set on prior versions.
+No compatability issues, unless this feature was later removed in which case a user would no longer be able to rely on this annotation.
+
+## Rejected alternatives
+
+The original implementation was going to use fields in the CR status, i.e. `Kafka.status.reconciled` but this would involve API changes, which after discussion with others seemed less preferably to simply using labels/annotations.
```

Comment:
```suggestion
The original implementation was going to use fields in the CR status, i.e. `Kafka.status.reconciled` but this would involve API changes, which after discussion with others seemed less preferable to simply using labels or annotations.
```